### PR TITLE
Update/vip utils phpcs issues

### PR DESCRIPTION
**Pull request by pschoffer**
## Description

Address #1855.

Because of our increased code standard requirements, more changes needed to be done to file. I tried to cover most of changes in unittests.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [x] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

Not sure really, try random clicking around and see if anything breaks.

### CODE REVIEW

File: vip-helpers/vip-utils.php [hunk 1]
```diff
@@ -1305,26 +1364,29 @@ function _wpcom_vip_include_plugin( $file ) {
...
+		'pre_include_variables' => 0,
+		'new_variables'         => 0,
+		'helper_file'           => 0,
+	);
 
 	// Let's find out what's new by comparing the current variables to the previous ones
 	$new_variables = array_diff_key( get_defined_vars(), $GLOBALS, $blacklist, $pre_include_variables );
 
 	// global each new variable
-	foreach ( $new_variables as $new_variable => $devnull )
```
Comment (pschoffer) [inline vip-helpers/vip-utils.php:1323]: This is the change that was needed for compliance reasons.

File: vip-helpers/vip-utils.php [hunk 2]
```diff
@@ -120,11 +121,12 @@ function vip_powered_wpcom( $display = 'text', $before_text = 'Powered by ' ) {
  * @return string
  */
 function vip_powered_wpcom_url() {
+	$utm_term = isset( $_SERVER['HTTP_HOST'] ) ? $_SERVER['HTTP_HOST'] : '';
```
Comment (GaryJones) [inline vip-helpers/vip-utils.php:124]: Could use [Null Coalescing](https://www.php.net/manual/en/migration70.new-features.php) here:

```suggestion
	$utm_term = $_SERVER['HTTP_HOST'] ?? '';
```

PHPCS shouldn't complain about that.

File: vip-helpers/vip-utils.php [hunk 3]
```diff
@@ -405,17 +413,21 @@ function vip_redirects( $vip_redirects_array = array(), $case_insensitive = fals
  * @param bool $append_old_uri Optional. If true, the full path past the match will be added to the new URL. Defaults to false.
  */
 function vip_substr_redirects( $vip_redirects_array = array(), $append_old_uri = false ) {
-	if ( empty( $vip_redirects_array ) )
+	if ( empty( $vip_redirects_array ) ) {
 		return;
+	}
 
+	$request_uri = isset( $_SERVER['REQUEST_URI'] ) ? $_SERVER['REQUEST_URI'] : '';
```
Comment (nickdaugherty) [inline vip-helpers/vip-utils.php:420]:
```suggestion
	$request_uri = $_SERVER['REQUEST_URI'] ) ?? '';
```

Guess we can use null coalescing here to match the other instance.

File: vip-helpers/vip-utils.php [hunk 4]
```diff
@@ -661,10 +684,14 @@ function wpcom_vip_is_feedservice_ua() {
...
 		// ASSUMPTION: you've already confirmed is_feed() b/f calling
 		// wpcom_vip_is_feedservice_ua
-			header( "X-Accel-Expires: 0" );
+			header( 'X-Accel-Expires: 0' );
 	}
 
-	return (bool) preg_match("/feedburner|feedvalidator|MediafedMetrics/i", $_SERVER["HTTP_USER_AGENT"]);
+
+	//phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
+	$http_user_agent = isset( $_SERVER['HTTP_USER_AGENT'] ) ? $_SERVER['HTTP_USER_AGENT'] : '';
```
Comment (nickdaugherty) [inline vip-helpers/vip-utils.php:692]:
```suggestion
	$http_user_agent = $_SERVER['HTTP_USER_AGENT'] ?? '';
```

File: vip-helpers/vip-utils.php [hunk 5]
```diff
@@ -1406,12 +1468,14 @@ function vip_is_jetpack_request() {
 		return false;
 	}
 
+	//phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
+	$http_user_agent = isset( $_SERVER['HTTP_USER_AGENT'] ) ? $_SERVER['HTTP_USER_AGENT'] : '';
```
Comment (nickdaugherty) [inline vip-helpers/vip-utils.php:1472]:
```suggestion
	$http_user_agent = $_SERVER['HTTP_USER_AGENT'] ?? '';
```

File: vip-helpers/vip-utils.php [hunk 6]
```diff
@@ -405,17 +413,21 @@ function vip_redirects( $vip_redirects_array = array(), $case_insensitive = fals
  * @param bool $append_old_uri Optional. If true, the full path past the match will be added to the new URL. Defaults to false.
  */
 function vip_substr_redirects( $vip_redirects_array = array(), $append_old_uri = false ) {
-	if ( empty( $vip_redirects_array ) )
+	if ( empty( $vip_redirects_array ) ) {
 		return;
+	}
 
+	$request_uri = $_SERVER['REQUEST_URI'] ) ?? '';
```
Comment (nickdaugherty) [inline vip-helpers/vip-utils.php:420]:
```suggestion
	$request_uri = $_SERVER['REQUEST_URI'] ?? '';
```
I left a closing parenthesis 🤷

**Review by nickdaugherty — APPROVED** [review]
So much great cleanup here, and I love the new tests 🙇 

We'll need to thoroughly test the plugin global variable changes (which I can help out with).

**Comment by nickdaugherty** [comment]
I did some sandbox testing and couldn't find any issues. The `$GLOBALS` change _should_ be fine. There's always the possibility of a weird edge case where it behaves differently, though.

**Comment by pschoffer** [comment]
Stacks-r1282.

**Pull request merged**